src/Tools/AnalyzerRunner/IncrementalAnalyzerRunner.cs

```diff
@@ -50,26 +48,13 @@ internal async Task RunAsync(Workspace workspace, CancellationToken cancellation
 
             var exportProvider = (IMefHostExportProvider)workspace.Services.HostServices;
 
-            var solutionSizeTracker = exportProvider.GetExports<ISolutionSizeTracker>().Single().Value;
-
             // This will return the tracker, since it's a singleton.
-            var analyzer = ((IIncrementalAnalyzerProvider)solutionSizeTracker).CreateIncrementalAnalyzer(workspace);
-            await analyzer.NewSolutionSnapshotAsync(workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
 
             var solutionCrawlerRegistrationService = (SolutionCrawlerRegistrationService)workspace.Services.GetRequiredService<ISolutionCrawlerRegistrationService>();
             solutionCrawlerRegistrationService.Register(workspace);
-            solutionCrawlerRegistrationService.WaitUntilCompletion_ForTestingPurposesOnly(workspace, ImmutableArray.Create(analyzer));
-
-            var size = solutionSizeTracker.GetSolutionSize(workspace, workspace.CurrentSolution.Id);
-            Console.WriteLine("Current solution size:\t" + size);
 
             if (_options.UsePersistentStorage)
             {
-                if (size <= 0)
-                {
-                    throw new InvalidOperationException("Solution size is too small; persistent storage is disabled.");
-                }
-
                 var persistentStorageService = workspace.Services.GetRequiredService<IPersistentStorageService>();
                 var persistentStorage = persistentStorageService.GetStorage(workspace.CurrentSolution);
                 if (persistentStorage is NoOpPersistentStorage)
```

Comment:
❔ Do we still have NoOpPersistentStorage? What is it used for?

Reply:
NoOp still occurs when the DB fails entirely (i.e. we can't even create it from sqlite's perspective) or we just have a Workspace host that doesn't support this at all.